src/abstractdataframe/show.jl

```diff
@@ -56,7 +56,7 @@ ourshow(io::IO, x::SHOW_TABULAR_TYPES; styled::Bool=false) =
     ourshow(io, summary(x), styled=styled)
 function ourshow(io::IO, x::Markdown.MD)
     r = repr(x)
-    len = min(something(findfirst(c->c=='\n', r), length(r) + 1) - 1, 50)
+    len = min(length(r, 1, something(findfirst(c->c=='\n', r), lastindex(r)+1)-1), 50)
```

Comment:
A small suggestion as `==(...)` is already in 1.0 if I remember correctly:
```suggestion
    # here we use the fact that `repr` produces `String` and `\n` is stored on 1 byte in UTF-8
    len = min(length(r, 1, something(findfirst(==('\n'), r), lastindex(r)+1)-1), 50)
```
also do you thing that 50 is reasonable? Maybe we can cut it down to e.g. 32? Not sure here - @ronisbr - any opinon here?

Reply:
@StefanKarpinski - I will write a blog post on string indexing in Base soon I think 😄, as it is really tricky which functions take byte indices and which take character indices.

Reply:
I'd be happy to change to 32 or ????.  I'll wait for a consensus.   I've made and tested the `==('\n')` change on 1.5.

Coming from a C/C++ background, I'm used to characters being 1 unsigned byte.  It took me a while to acknowledge that, in Julia, they are 4 bytes.  However, when you assign them to a string, all of sudden they become 1,2,3 or 4 bytes depending.  Fine.  That is just UTF-8 encoding.  Then when I compare the unsigned byte at index i in a string to `'\n'` it sort of looks like a single byte comparison.  But it isn't.  When I do 
```julia
julia> s="∫αγ∞1∫αγ∞2"
"∫αγ∞1∫αγ∞2"

julia>  findfirst(==('∞'),"∫αγ∞1∫αγ∞2")
8

julia> s[8]
'∞': Unicode U+221E (category Sm: Symbol, math)
```
I'm clearly comparing against a 4-byte item but get an index into the 1 byte array.  So `findfirst(...)` must convert each valid character to a 4-byte and then perform the comparison but pass back the 1-byte index.  All this back and forth is a little mind-bending.  Particularly, when it isn't clear from the function names which indexing mode they operate in.  To a novice, `str[i]` would seem like it should return the `i-th` 4-byte character not the character at byte index  `i` if there is one or an error otherwise.   You just have to know.  A blog post that explained the underlying logic would really help.

That's not to say that my blunder wasn't foolish. :smiley:

Reply:
`==('\n')` for sure works on 1.5 and I think it works on 1.0, I just do not remember when it what introduced. Let us just pass it through CI :). And I propose to change the default width to 32 then, as wider output in the REPL is not very useful I think.

> That's not to say that my blunder wasn't foolish. 😃

Actually what I thought was that it was not foolish, but that the documentation of strings is not explicit enough here (I know we highlight the differences in indexing in https://docs.julialang.org/en/latest/manual/strings/#Unicode-and-UTF-8-1, but we do not cover in detail how functions behave, e.g. when you read a docsting for `findfirst` it does not say what kind of index you will get).